highfidelity/hifi-content
# Applause Application for Zaru

This feature will allow users who enter into Zaru to applaud in desktop and HMD mode. In desktop mode, the user will get a button on the screen and be able to clap with a keyboard key press. In HMD mode, the user will be able to shake their hand controllers to allow them to generate applause sounds and heart particles. 

**Test Plan - App Setup**
1. In a domain where you have rez permissions, add a cube and put https://hifi-content.s3-us-west-1.amazonaws.com/liv/Prototypes/Clapping-V4/loadApplauseApp.js into the script URL field. 

Confirm that the button appears on your HUD / tablet. 

2. Reload scripts (CTRL + R)

Confirm that you do not get multiple instances of the button

3. Leave the domain and return

Confirm that the button is removed when you exit your domain, and returns when you come back to the domain close to where the cube is. 

**Test Plan - Desktop Mode**
1. Click on the button for the applause app on your HUD. 

Confirm that you are presented with a UI that explains the app and has a toggle to enable the applause setting. 

2. Toggle on the applause setting

Confirm that you see a button with instructions, a toggle for using the particle effects, and a new overlay button has been added to the bottom left corner of your screen. 

Confirm that when you click the buttons or press the 't' key on your keyboard that you see applause. Configure the app and open and close it to check that states remain the same when enabling and disabling the components. Toggling off 'use particles' should keep the hearts from showing. Confirm that your avatar scaling changes and that the particles still come from the position of your avatar's head. You should hear audio playing for the clapping.

**Test Plan - HMD Mode**

1. Open up the tablet app for the applause app in your HMD.

Confirm that you see the tablet UI and a toggle. 

2. Toggle on the applause setting.

Confirm that you see a gif that shows you how to move your hands.

3. Try clapping.

Confirm that as you bring your hands close while applause is enabled, your hands flatten out and return to their usual state when pulled away. Confirm that the desktop button overlay is not present around you somewhere in world. Confirm that the particles follow your hands and that the audio plays. 

4. Switch between desktop and VR mode. Confirm that the app cleans up appropriately between swapping (e.g. your hands do not stay flattened, your overlay returns) and maintains your configurations from the app.
